lib/manageiq/api/common/headers.rb

```diff
@@ -0,0 +1,44 @@
+module HeaderScope
+  thread_mattr_accessor :current
+end
+
+module ManageIQ
+  module API
+    module Common
+      module Headers
+
+        def self.current=(val)
+          validate_headers(val)
+          HeaderScope.current = val
+        end
+
+        def self.current
+          HeaderScope.current
+        end
+
+        def self.decode(headers, key)
+          validate_headers(headers)
+          val = headers.instance_variable_get(:@req).stringify_keys
+          JSON.parse(Base64.decode64(val[key]))
```

Comment:
@syncrou 
From the headers how do you know which ones have to be base64 decoded and which ones have to be left alone.
If we get passed in a User-Agent in the HTTP header that data is not base64 encoded. Do we need to maintain a list of keys which are base64 encoded and which ones aren't.

Reply:
That's a good question.   I'm not sure if that is an implementation question for the microservice or something the common gem should handle.